### PR TITLE
Release MU01 Phase 1 on Celo

### PR DESCRIPTION
**Pull request by bowd**
### Description

This PR deploys MU01 contract to Celo (mainnet) and introduces tooling changes around mainnet deployments.
We added scripts to build a `json` proposal representation compatible with celo tooling, which will be used to submit to the celo-org/governance repo.

### Other changes

- Fixes a config issue around minimum reports
- Rename verifyBytecode to verify-bytecode to keep with naming conventions 

### Tested

Tested on a fork

### Related issues

N/A

### Backwards compatibility

Yes

### Documentation

- Addresses added to docs
- Verification docs updated

### CODE REVIEW

File: script/upgrades/dependencies.json [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {
-  "42220": {},
+  "42220": {
+    "BridgedUSDC": "0xEB466342C4d449BC9f53A865D5Cb90586f405215",
```
Comment (nvtaveras) [inline script/upgrades/dependencies.json:3]: sucks that this contract is not verified

File: script/upgrades/dependencies.json [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+    "USDCUSDRateFeedAddr": "0xA1A8003936862E7a15092A91898D69fa8bCE290c",
+    "PartialReserveMultisig": "0x87647780180B8f55980C7D3fFeFe08a9B29e9aE1"
```
Comment (nvtaveras) [inline script/upgrades/dependencies.json:5]: how can I verify this address?

File: bin/setup.sh [hunk 3]
```diff
@@ -47,7 +57,7 @@ parse_upgrade () { # $1: upgrade
 
 forge_script () { # $1: script name, $2: script file path
     echo "=================================================================="
-    echo "🔥 Running $1"
+    echo "🏃🏼 Running $1"
```
Comment (nvtaveras) [inline bin/setup.sh:60]: pun intended?

File: bin/setup.sh [hunk 4]
```diff
@@ -23,6 +23,16 @@ parse_network () { # $1: network
             CHAIN_ID=44787
             export FOUNDRY_PROFILE=alfajores-deployment
             ;;
+        "celo")
+            # RPC_URL=$CELO_RPC_URL
+            RPC_URL="http://127.0.0.1:8545"
```
Comment (nvtaveras) [inline bin/setup.sh:28]: seems like you forgot to change this back?

File: foundry.toml [hunk 5]
```diff
@@ -45,8 +45,9 @@ libraries = [
 ]
 
 [profile.celo-deployment]
-# TODO: Add libraries during deployment
 libraries = [
+    "lib/mento-core/contracts/common/linkedlists/AddressLinkedList.sol:AddressLinkedList:0x6200F54D73491d56b8d7A975C9ee18EFb4D518Df",
```
Comment (nvtaveras) [inline foundry.toml:49]: why is this contract not verified?

File: bin/cgp-diff.sh [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+##############################################################################
+# Script for passing a Celo Governance Proposal on a tesnet.
```
Comment (nvtaveras) [inline bin/cgp-diff.sh:4]:
```suggestion
# Script for passing a Celo Governance Proposal on a testnet.
```

File: bin/cgp-diff.sh [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+# Usage: ./bin/cgp-diff.sh 
+#               -n <baklava|alfajores>  -- network to pass the proposal on
+#               -p <proposal_id>        -- proposal ID
+# Example: ./bin/cgp-pass.sh -n baklava -p 79
+##############################################################################
+
+source "$(dirname "$0")/setup.sh"
+
+NETWORK=""
+PROPOSAL_ID=""
```
Comment (nvtaveras) [inline bin/cgp-diff.sh:14]: remove?

File: bin/cgp-diff.sh [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+# Get the proposal transactions count
+TX_COUNT_ALPHA=$(cast call $GOVERNANCE "getProposal(uint256)(address,uint256,uint256,uint256,string)" $PROPOSAL_ID_ALPHA --rpc-url $RPC_URL | tail -2 | head -1)
+TX_COUNT_BETA=$(cast call $GOVERNANCE "getProposal(uint256)(address,uint256,uint256,uint256,string)" $PROPOSAL_ID_BETA --rpc-url $RPC_URL | tail -2 | head -1)
+
+if [ "$TX_COUNT_ALPHA" != "$TX_COUNT_BETA" ]; then
+    echo "🚨 Proposal transactions count is different"
+    exit 1
+fi
+
+echo $TX_COUNT_ALPHA
```
Comment (nvtaveras) [inline bin/cgp-diff.sh:48]: remove?

File: bin/cgp-diff.sh [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+for ((i = 0; i < $TX_COUNT_ALPHA; ++i)); do
```
Comment (nvtaveras) [inline bin/cgp-diff.sh:50]: do you need the double parens `((` here?

File: script/upgrades/dependencies.json (hunk 2 above)
Comment (bowd) [reply script/upgrades/dependencies.json:5]: https://docs.mento.org/mento/developers/deployments/addresses

and trust

File: script/upgrades/dependencies.json (hunk 1 above)
Comment (bowd) [reply script/upgrades/dependencies.json:3]: 👍 agreed.

File: foundry.toml (hunk 5 above)
Comment (bowd) [reply foundry.toml:49]: Verified it now 🎉

File: bin/cgp-yaml-to-json.sh [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+#    anvil --fork-url $CELO_RPC_URL
+# 3. Submit the CGP to the forked node, and get the proposal ID:
+#    yarn cgp -n celo -u MU01 -p <phase> -f 
+# 4. Get the yaml output of the CGP: 
+#    celocli governance:show --proposalID <id> --node http://127.0.0.1:8545 > cgp.yaml
+# 5. Transform to json input: 
+#    yarn cgp:yaml-to-json cgp.yaml > cgp.json
+#
+# At this point you have a a cgp.json that's compatible with the celocli tooling 
+# (pending the celocli tooling being updated to support the new CGP format).
```
Comment (tobikuhlmann) [inline bin/cgp-yaml-to-json.sh:34]: semms like your monorepo celocli PR was merged? Then this line could be removed

File: bin/cgp.sh [hunk 11]
```diff
@@ -16,19 +17,26 @@ NETWORK=""
...
         s) SIMULATE=true;;
+        f) USE_FORK=true;;
     esac
 done
 
 parse_network "$NETWORK"
 parse_upgrade "$UPGRADE"
 
+if [ "$USE_FORK" = true ] ; then
+    RPC_URL="http://127.0.0.1:8545"
```
Comment (tobikuhlmann) [inline bin/cgp.sh:36]: could be helpful to add a line to the comments above that this requires a anvil node at localhost

File: script/upgrades/dependencies.json [hunk 12]
```diff
@@ -1,5 +1,9 @@
 {
-  "42220": {},
+  "42220": {
+    "BridgedUSDC": "0xEB466342C4d449BC9f53A865D5Cb90586f405215",
+    "USDCUSDRateFeedAddr": "0xA1A8003936862E7a15092A91898D69fa8bCE290c",
```
Comment (tobikuhlmann) [inline script/upgrades/dependencies.json:4]: what is this EOA address?

Comment (bowd) [reply script/upgrades/dependencies.json:4]: This is the RateFeedID of the USDCUSD oracle rate. It's the target address used for reporting. It's that `address(uint160(keccak256("USDCUSD")))` thing that we couldn't find a good place to put in the docs 😂

File: bin/cgp-diff.sh (hunk 9 above)
Comment (bowd) [reply bin/cgp-diff.sh:47]: yes :-]

File: script/upgrades/dependencies.json (hunk 12 above)
Comment (tobikuhlmann) [reply script/upgrades/dependencies.json:4]: aah now it makes sense :D maybe call it `...RateFeedID` instead of `RateFeedAddr`?

**Review by tobikuhlmann — APPROVED** [review]
now it looks gooddd to me

**Comment by bowd** [comment]
@nvtaveras can you take another look, your requested changes have been tackled

**Comment by nvtaveras** [comment]
> @nvtaveras can you take another look, your requested changes have been tackled

seems like one of my comments was still not addressed, other than that LGTM

**Pull request merged**